src/common/transformations/src/transformations/common_optimizations/convert_compressed_to_mixed_precision.cpp

```diff
@@ -20,13 +21,13 @@ bool ov::pass::ConvertCompressedToMixedPrecision::run_on_model(const std::shared
         return false;
 
     Manager manager(get_pass_config());
-    // Mark nodes in ShapeOf subgraphs with disable_fp16_compression rt_info to keep them in FP32 precision
-    REGISTER_PASS(manager, MarkPrecisionSensitiveShapeOfSubgraphs)
+    REGISTER_PASS(manager, MarkSugraphsToKeepInMixedPrecision)
+    REGISTER_PASS(manager, AlignMixedFP32FP16Types)
```

Comment:
can we combine this two transformations into one? These two transformations operate together and it make sense to put them into one pass where in the first step you perform marking and after you align types.

Reply:
They are operating together and are already called together inside in a single pass here inside `ConvertCompesedToMixed..` :)

You think that it's potentially dangerous if someone calls only one of them? If someone will call separately only `MarkSugraphsToKeepInMixedPrecision` or `AlignMixedFP32FP16Types` this will cause no effect and damage at all. If you mark graph but do not call ConvertPrecision with special argument then this markings are ignored.

Misuse can happen if someone calls them without ConvertPrecision in one place and other transformations modify inside precision sensitive nodes, or some Converts are deleted and calls Convert somewhere long after that. But putting them together will not prevent from such a misuse at all.

And we already have transformations where marking is done in one transformation and using that marking in another place, they are not even placed one after another in CommonOptimizations e.g. here `MarkPrecisionSensitiveDivides` and `ConvertDivide` are scattered over the CommonOptimization
https://github.com/openvinotoolkit/openvino/pull/9561/files#diff-2ed72810580174cd45592b5336384f0dad9c9dd47904be56373376d057fd022f